testonly/hammer/hammer.go

```diff
@@ -165,8 +166,26 @@ func (c MapConfig) String() string {
 		c.MapID, c.EPBias, c.Operations, c.EmitInterval, c.RetryErrors)
 }
 
+// SetULimit calls `ulimit -n` to set the maximum number of open files.
+// See https://wilsonmar.github.io/maximum-limits/
+func SetULimit(uLimit uint64) error {
+	var rLimit syscall.Rlimit
+	err := syscall.Getrlimit(syscall.RLIMIT_NOFILE, &rLimit)
+	if err != nil {
+		return err
+	}
+	if uLimit > rLimit.Max {
+		return fmt.Errorf("`ulimit -n %v -S` failed because it is greater than the hard limit %v", uLimit, rLimit.Max)
+	}
+	rLimit.Cur = uLimit
```

Comment:
If `uLimit` is smaller than `rLimit.Cur` already is, do you really want to reduce it?

Reply:
I think yes, for simplicity's sake. This only sets it for the current process, so there's no harm in doing so. If the caller is trying to set it to be too low, the caller should find out.